react-leaflet peer dependency

Possible to loosen peer dependencies on react-leaflet to support all 4.x.x series? It's currently [4.2.1](https://github.com/PaulLeCam/react-leaflet/releases/) 

```
npm ERR! Conflicting peer dependency: react-leaflet@4.2.1
npm ERR! node_modules/react-leaflet
npm ERR!   peer react-leaflet@"^4.0.2" from react-leaflet-geoman-v2@1.0.0
npm ERR!   node_modules/react-leaflet-geoman-v2
npm ERR!     react-leaflet-geoman-v2@"^1.0.0" from the root project
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-leaflet-geoman-v2",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "React wrapper for the leaflet-geoman plugin",
   "repository": "https://github.com/TurtIeSocks/react-leaflet-geoman",
   "author": "TurtIeSocks <58572875+TurtIeSocks@users.noreply.github.com>",
@@ -15,9 +15,6 @@
     "start": "vite",
     "build": "vite build",
     "build:ts": "tsc --project tsconfig.build.json"
-  },
-  "dependencies": {
-    "use-deep-compare-effect": "^1.8.1"
   },
   "devDependencies": {
     "@geoman-io/leaflet-geoman-free": "^2.14.2",
@@ -40,6 +37,6 @@
     "@geoman-io/leaflet-geoman-free": "^2.14.2",
     "react": "^17.0.0 || ^18.0.0",
     "react-dom": "^17.0.0 || ^18.0.0",
-    "react-leaflet": "^4.0.2"
+    "react-leaflet": "^4"
   }
 }

--- a/src/GeomanControls.ts
+++ b/src/GeomanControls.ts
@@ -2,7 +2,6 @@ import '@geoman-io/leaflet-geoman-free'
 import { useLayoutEffect, useEffect, useState } from 'react'
 import { useLeafletContext } from '@react-leaflet/core'
 import type { LayerGroup } from 'leaflet'
-import useDeepCompareEffect from 'use-deep-compare-effect'
 
 import type { GeomanProps } from './types'
 import { reference, layerEvents, globalEvents, mapEvents } from './events'
@@ -56,7 +55,7 @@ export default function GeomanControls({
     if (mounted) map.pm.setPathOptions(pathOptions)
   }, [pathOptions, mounted])
 
-  useDeepCompareEffect(() => {
+  useEffect(() => {
     // set global options
     if (mounted)
       map.pm.setGlobalOptions({ layerGroup: container, ...globalOptions })

